action.yml

```diff
@@ -0,0 +1,60 @@
+name: 'NPM Version Incrementer'
+description: 'Increments the version of an NPM package based on the branch.'
+outputs:
+  branch-name:
+    description: 'The current branch name.'
+    value: ${{ steps.get-branch-name.outputs.branch-name }}
+  current-version:
+    description: 'The current version of the package.'
+    value: ${{ steps.get-current-version.outputs.current-version }}
+  new-version:
+    description: 'The new version of the package.'
+    value: ${{ steps.get-new-version.outputs.new-version }}
+runs:
+  using: "composite"
+  steps:
+    - id: get-branch-name
+      shell: bash
+      run: |
+          # Get the current branch name
+          BRANCH=$(git branch --show-current)
+          echo "Current branch is - $BRANCH"
+          echo "branch-name=$BRANCH" >> $GITHUB_OUTPUT
+
+    - id: get-current-version
+      shell: bash
+      run: |
+          # Get the current version number
+          CURRENT_VERSION=$(jq -r '.version' package.json)
+          echo "Current version is - $CURRENT_VERSION"
+          echo "current-version=$CURRENT_VERSION" >> $GITHUB_OUTPUT
+
+    - id: npm-version-increment
+      shell: bash
+      run: |
+          BRANCH=${{ steps.get-branch-name.outputs.branch-name }}
+          echo "Current branch is - $BRANCH"
+          # Check if it's the master branch or not
+          if [ "$BRANCH" == "master" ] || [ "$BRANCH" == "main" ]; then
+            # For the master branch, bump a minor version
+            echo "Bumping minor version"
+            npm version minor
+          else
+            # For non-master branches, create a pre-release version
+            echo "Bumping patch version"
+            npm version patch
+          fi
+
+    - id: get-new-version
+      shell: bash
+      run: |
+          # Get the updated version number
+          NEW_VERSION=$(jq -r '.version' package.json)
+          echo "New version is - $NEW_VERSION"
+          echo "new-version=$NEW_VERSION" >> $GITHUB_OUTPUT
+
+    - id: git-commit
+      shell: bash
+      run: |
+          # Push the new tag to GitHub
+          git push --follow-tags origin $BRANCH
```

Comment:
The following two settings are useful in VS Code:
![image](https://github.com/Aqovia/npm-version-increment/assets/46093/3c2fb6cc-f17b-4d71-b66b-90d88e00e6c1)

![image](https://github.com/Aqovia/npm-version-increment/assets/46093/ae897378-ade1-4271-b3a6-d0b77ba71d72)


So files are ending with a new line. It helps to have better commits.

Reply:
I'll get this added to my VSCode.